src/lib.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017, Nick Stevens <nick@bitcurry.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/license/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Composable first-order predicate functions.
+//!
+//! This library implements an interface to "predicates" - boolean-valued
+//! functions of one argument. This allows combinatorial logic to be created and
+//! assembled at runtime and then used one or more times for testing values.
+//! This sort of object is really useful when creating filters and checks that
+//! can be changed at runtime with user interaction - it allows a clean
+//! separation of concerns where the configuration code can be used to build up
+//! a predicate, and then that predicate can be given to the code that does the
+//! actual filtering without the filtering code knowing anything about user
+//! configuration. See the examples for how this can work.
+//!
+//! ## Installation
+//!
+//! Add this to your `Cargo.toml`:
+//!
+//! ```toml
+//! [dependencies]
+//! predicates = "0.1"
+//! ```
+//!
+//! ## Examples
+//!
+//! A few different examples of how predicates might be used:
+//!
+//! ```
+//! use predicates::{predicate, Predicate};
+//!
+//! // The simplest predicates are `always()` and `never()`, which always returns
+//! // `true` and always returns `false`, respectively. The values are simply
+//! // ignored when testing against these predicates:
+//! let always_true = predicate::always();
+//! assert!(always_true.test(&5));
```

Comment:
Yeah, it just makes it more obvious than a `!` that your eyes can skip over easily.